src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java

```diff
@@ -333,12 +341,34 @@ protected void initializeInternal(final DataManager manager,
             max[i] = (input.getDataLength() * log2(input.getDataLength())) * Math.max(gFactor, sFactor);
         }
         
+        if (config.isPrivacyModelSpecified(DataDependentEDDifferentialPrivacy.class)) {
+            
+            // Store minimal size of equivalence classes
+            DataDependentEDDifferentialPrivacy dpCriterion = config.getPrivacyModel(DataDependentEDDifferentialPrivacy.class);
+            k = (double)dpCriterion.getK();
+            
+            // Store root values of generalization hierarchies or null if no single root value exists
+            rootValues = new Integer[hierarchies.length];
```

Comment:
Why is this array of type Integer[]. int[] is sufficient. You can encode a missing root value as -1, for example.